### PR TITLE
test(dialog-full-screen): refactor ts

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

- Rename the `dialog-full-screen.cy.js` to `dialog-full-screen.cy.tsx` file in `./cypress/components/dialog-full-screen/` folder.
- Refactor tests to Typescript.  

### Current behaviour

`Dialog-full-screen` tests are in js not ts.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the `dialog-full-screen.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other `*.cy.tsx` files have regressed

### CODE REVIEW

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 1]
```diff
@@ -293,13 +287,19 @@ context("Testing DialogFullScreen component", () => {
         .should("have.css", "padding-right", "0px")
         .should("have.css", "padding-bottom", "0px");
     });
+
+    // skip this test for now as FE-6053 has not been fixed yet
+    it.skip("should render DialogFullScreen component with DisableClose prop", () => {
+      CypressMountWithProviders(<DialogFullScreenComponent disableClose />);
+      closeIconButton().should("be.disabled").and("have.attr", "disabled");
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:294]: we suppose to add that if we press/trigger this button the `DialogFullScreen` should remain opened

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 2]
```diff
@@ -293,13 +287,19 @@ context("Testing DialogFullScreen component", () => {
...
+    });
   });
 
   describe("should check accessibility for Dialog Full Screen", () => {
     it("should check accessibility for default Dialog Full Screen component", () => {
       CypressMountWithProviders(<DefaultDocsStory />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:302]: could we avoid using this `return` here?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 3]
```diff
@@ -308,7 +308,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen with complex example", () => {
       CypressMountWithProviders(<WithComplexExample />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:311]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 4]
```diff
@@ -317,7 +317,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen with disabled content padding", () => {
       CypressMountWithProviders(<WithDisableContentPadding />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:320]: return

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 5]
```diff
@@ -326,7 +326,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen component with header children", () => {
       CypressMountWithProviders(<WithHeaderChildren />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:329]: return

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 6]
```diff
@@ -335,7 +335,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen component with help", () => {
       CypressMountWithProviders(<WithHelp />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:338]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 7]
```diff
@@ -344,7 +344,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen component with hideable header children", () => {
       CypressMountWithProviders(<WithHideableHeaderChildren />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:347]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 8]
```diff
@@ -353,7 +353,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen component with box", () => {
       CypressMountWithProviders(<WithBox />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:356]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 9]
```diff
@@ -368,7 +368,7 @@ context("Testing DialogFullScreen component", () => {
         .then(() => cy.checkAccessibility());
       closeIconButton().click();
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:371]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 10]
```diff
@@ -377,7 +377,7 @@ context("Testing DialogFullScreen component", () => {
     it("should check accessibility for default Dialog Full Screen component with other focusable containers", () => {
       CypressMountWithProviders(<OtherFocusableContainers />);
 
-      buttonDataComponent()
+      return buttonDataComponent()
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:380]: same here

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by stephenogorman — DISMISSED** [review]

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 11]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import PropTypes from "prop-types";
+import { DialogFullScreenProps } from "components/dialog-full-screen";
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:2]: could you update the path?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 12]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
 
     it("should check accessibility for default Dialog Full Screen component with header children", () => {
       CypressMountWithProviders(<WithHeaderChildren />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
+      cy.checkAccessibility();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:325]: shall we add something to be sure that Dialog is opened? Like the crossIcon is visible, or something similar?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 13]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
 
   describe("should check accessibility for Dialog Full Screen", () => {
     it("should check accessibility for default Dialog Full Screen component", () => {
       CypressMountWithProviders(<DefaultDocsStory />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:303]: shall we add something to be sure that Dialog is opened? Like the crossIcon is visible, or something similar?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 14]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen with complex example", () => {
       CypressMountWithProviders(<WithComplexExample />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:310]: shall we add something to be sure that Dialog is opened? Like the crossIcon is visible, or something similar?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 15]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen with disabled content padding", () => {
       CypressMountWithProviders(<WithDisableContentPadding />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:317]: shall we add something to be sure that Dialog is opened? Like the crossIcon is visible, or something similar?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 16]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen component with help", () => {
       CypressMountWithProviders(<WithHelp />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:331]: shall we add something to be sure that Dialog is opened? Like the crossIcon is visible, or something similar?

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 17]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen component with hideable header children", () => {
       CypressMountWithProviders(<WithHideableHeaderChildren />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:338]: and here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 18]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen component with box", () => {
       CypressMountWithProviders(<WithBox />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:345]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 19]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
-        .then(() => cy.checkAccessibility());
+        .click();
+      cy.checkAccessibility();
       closeIconButton().click();
 
-      buttonDataComponent()
-        .contains("Open Demo using autoFocus")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open Demo using autoFocus").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:358]: same here

File: cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx [hunk 20]
```diff
@@ -293,94 +287,83 @@ context("Testing DialogFullScreen component", () => {
...
     });
 
     it("should check accessibility for default Dialog Full Screen component with other focusable containers", () => {
       CypressMountWithProviders(<OtherFocusableContainers />);
 
-      buttonDataComponent()
-        .contains("Open DialogFullScreen")
-        .click()
-        .then(() => cy.checkAccessibility());
+      buttonDataComponent().contains("Open DialogFullScreen").click();
```
Comment (DlgSHi) [inline cypress/components/dialog-full-screen/dialog-full-screen.cy.tsx:365]: and here

**Review by tomdavies73 — APPROVED** [review]

**Pull request merged**